src/null-safety/index.md

```diff
@@ -1,39 +1,247 @@
+then some of the null-safety syntax might look familiar.
+That's by design: the Dart language aims to be unsurprising.
+
+
+## Creating and using variables (?, !, late)
+
+Use `?`, `!`, and `late` to tweak how the Dart analyzer treats your code.
+
+Here’s an example of declaring a **non-nullable integer variable**
+(assuming you’ve opted into null safety):
+
+```dart
+int aRealInt; // Can't be used until it's initialized to an integer.
+```
+
+If the variable _can_ have the value `null`,
+**add  `?`** to its type declaration:
+
+```dart
+int? aNullableInt = null;
+```
+
+When using a value that you know isn’t null but that has a nullable type,
+**add `!`** to tell the Dart analyzer that the value isn’t null:
+
+```dart
+int? a = 2;
+int b = a!; // The ! temporarily changes `a` from int? to int.
+```
+
+If you know that a non-nullable value will be
+initialized to a non-null value before it's used,
+but the Dart analyzer thinks otherwise,
+**add the `late` keyword** to the type:
+
+```dart
+late int aRealInt;
+```
+
+{% comment %}
+{{ site.alert.tip }}
+  Avoid using `late final` unless you really need it.
+  For example, don't use `late final` in functions,
+  and in constructors prefer using an [initializer list][].
+{{ site.alert.end }}
+
+[initializer list]: /guides/language/language-tour#initializer-list
+{% endcomment %}
+
+
+## Understanding list, set, and map types
+
+Lists, sets, and maps are commonly used collection types in Dart programs,
+so you need to know how they interact with null safety.
+Here are some examples of how Dart code uses these collection types:
+
+* Flutter layout widgets such as [`Column`][]
+  often have a `children` property that’s a
+  `List` of `Widget` objects.
+* The [Veggie Seasons][] sample uses a `Set` of `VeggieCategory` to
+  store a user’s food preferences.
+* The [GitHub Dataviz][] sample has a `fromJson()`method that
+  creates an object from JSON data that’s supplied in a
+  `Map<String, dynamic>`.
+
+[`Column`]: https://api.flutter.dev/flutter/widgets/Column-class.html
+[Veggie Seasons]: https://github.com/flutter/samples/tree/master/veggieseasons
+[GitHub Dataviz]: https://github.com/flutter/samples/tree/master/web/github_dataviz
+
+
+### List and set types {#list-and-set-types}
+
+When you’re declaring the type of a list or set, think about what can be null. The following table shows the possibilities for a list of strings
+if you opt into null safety.
+
+{% assign yes = '<b>Yes</b>' %}
+
+|------------------+----------------------------------+--------------------------+------------|
+|Type              | Can an item (string)<br>be null? | Can the list<br>be null? | Description|
+|------------------|----------------------------------|--------------------------|------------|
+| `List<String>`   | No      | No      | A non-null list that contains<br> non-null strings       |
+| `List<String>?`  | No      | {{yes}} | A non-null list that contains<br> strings that **might be null** |
```

Comment:
The description is wrong. It should be swapped with the next line.

Reply:
Yikes! Thanks for catching that. Btw, I've decided to swap the boolean columns so the bold explanation corresponds with the bold **yes**es. I think I got it right this time.